cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/TopicReplicaDistributionGoal.java

```diff
@@ -425,7 +427,7 @@ private boolean rebalanceByMovingReplicasOut(Broker broker,
 
     int numReplicasOfTopicInBroker = broker.numReplicasOfTopicInBroker(topic);
 
-    for (Replica replica : replicasToMoveOut(clusterModel, broker, topic)) {
+    for (Replica replica : replicasToMoveOut(clusterModel, broker, topic, optimizationOptions)) {
```

Comment:
As a general pattern, we seem to double filter replicas (once to select them, once again to check whether `shouldExclude(...)`) to move in/out. I am not sure if this is necessary / optimal under different cases.

Can we create an issue to investigate whether we should/can merge this logic cleanly, and evaluate the impact on optimization performance?

Reply:
Agree, create PR #802 to track it.